src/search.rs

```diff
@@ -337,294 +462,303 @@ mod tests {
-        let index = setup_test_index(&client, "test_query_offset").await;
-
-        let results: SearchResults<Document> =
-            index.search().with_offset(6).execute().await.unwrap();
-        assert_eq!(results.hits.len(), 4);
-
-        client.delete_index("test_query_offset").await.unwrap();
+        with_test_index("test_query_offset", |index| async move {
+            let results: SearchResults<Document> =
+                index.search().with_offset(6).execute().await.unwrap();
+            assert_eq!(results.hits.len(), 4);
+        })
+        .await
     }
 
     #[async_test]
     async fn test_query_filter() {
-        let client = Client::new("http://localhost:7700", "masterKey");
-        let index = setup_test_index(&client, "test_query_filter").await;
-
-        let results: SearchResults<Document> = index
-            .search()
-            .with_filter("value = \"The Social Network\"")
-            .execute()
-            .await
-            .unwrap();
-        assert_eq!(results.hits.len(), 1);
+        with_test_index("test_query_filter", |index| async move {
+            let results: SearchResults<Document> = index
+                .search()
+                .with_filter("value = \"The Social Network\"")
+                .execute()
+                .await
+                .unwrap();
+            assert_eq!(results.hits.len(), 1);
+
+            let results: SearchResults<Document> = index
+                .search()
+                .with_filter("NOT value = \"The Social Network\"")
+                .execute()
+                .await
+                .unwrap();
+            assert_eq!(results.hits.len(), 9);
+        })
+        .await
+    }
 
-        let results: SearchResults<Document> = index
-            .search()
-            .with_filter("NOT value = \"The Social Network\"")
-            .execute()
-            .await
-            .unwrap();
-        assert_eq!(results.hits.len(), 9);
+    #[async_test]
+    async fn test_query_filter_and() {
+        with_test_index("test_query_filter_and", |index| async move {
+            let results: SearchResults<Document> = index
+                .search()
+                .with_filter(vec![
+                    "NOT value = \"The Social Network\"",
+                    "kind = \"title\"",
+                ])
+                .execute()
+                .await
+                .unwrap();
+            assert_eq!(results.hits.len(), 7);
+        })
+        .await
+    }
 
-        client.delete_index("test_query_filter").await.unwrap();
+    #[async_test]
+    async fn test_query_filter_andor() {
+        with_test_index("test_query_filter_and", |index| async move {
+            let results: SearchResults<Document> = index
+                .search()
+                .with_filter(vec![
+                    vec!["NOT value = \"The Social Network\""],
+                    vec!["kind = \"title\"", "kind = \"not title\""],
```

Comment:
FYI you could simplify readabilty by using the `r#` operator.
```suggestion
                    vec![r#"kind = "title"", "kind = "not title""#],
```